File: lib/features/billers/ui/search_screen_data.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'biller_list_item_data.dart';

part 'search_screen_data.freezed.dart';

@freezed
sealed class SearchScreenData with _$SearchScreenData {
  const factory SearchScreenData.idle() = SearchIdle;
  const factory SearchScreenData.searching() = SearchSearching;
  const factory SearchScreenData.empty(String query) = SearchEmpty;
  const factory SearchScreenData.error(String query) = SearchError;
  const factory SearchScreenData.results({
    required List<BillerListItemData> billers,
  }) = SearchResults;
}

File: app/src/main/java/com/billpayments/features/billers/ui/SearchScreenData.kt
package com.billpayments.features.billers.ui

import androidx.compose.runtime.Immutable

@Immutable
sealed interface SearchScreenData {
    data object Idle : SearchScreenData
    data object Searching : SearchScreenData
    data class Empty(val query: String) : SearchScreenData
    data class Error(val query: String) : SearchScreenData
    data class Results(val billers: List<BillerListItemData>) : SearchScreenData
}
